src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -85,77 +85,30 @@ class PythonSetup(Subsystem):
         ),
         advanced=True,
     )
-    requirement_constraints = FileOption(
-        "--requirement-constraints",
-        default=None,
-        help=softwrap(
-            """
-            When resolving third-party requirements for your own code (vs. tools you run),
-            use this constraints file to determine which versions to use.
-
-            This only applies when resolving user requirements, rather than tools you run
-            like Black and Pytest. To constrain tools, set `[tool].lockfile`, e.g.
-            `[black].lockfile`.
-
-            See https://pip.pypa.io/en/stable/user_guide/#constraints-files for more
-            information on the format of constraint files and how constraints are applied in
-            Pex and pip.
-
-            Mutually exclusive with `[python].enable_resolves`.
-            """
-        ),
-        advanced=True,
-        mutually_exclusive_group="lockfile",
-    )
-    resolve_all_constraints = BoolOption(
-        "--resolve-all-constraints",
-        default=True,
-        help=softwrap(
-            """
-            If enabled, when resolving requirements, Pants will first resolve your entire
-            constraints file as a single global resolve. Then, if the code uses a subset of
-            your constraints file, Pants will extract the relevant requirements from that
-            global resolve so that only what's actually needed gets used. If disabled, Pants
-            will not use a global resolve and will resolve each subset of your requirements
-            independently.
-
-            Usually this option should be enabled because it can result in far fewer resolves.
-
-            Requires [python].requirement_constraints to be set.
-            """
-        ),
-        advanced=True,
-    )
     enable_resolves = BoolOption(
         "--enable-resolves",
         default=False,
         help=softwrap(
             """
-            Set to true to enable the multiple resolves mechanism. See
-            `[python].resolves` for an explanation of this feature.
+            Set to true to enable lockfiles for user code. See `[python].resolves` for an
+            explanation of this feature.
 
-            Warning: the `generate-lockfiles` goal does not yet work if you have VCS (Git)
-            requirements and local requirements. Support is coming in a future Pants release. You
-            can still use multiple resolves, but you must manually generate your lockfiles rather
-            than using the `generate-lockfiles` goal, e.g. by running `pip freeze`. Specifically,
-            set up `[python].resolves` to point to your manually generated lockfile paths, and
-            then set `[python].resolves_generate_lockfiles = false` in `pants.toml`.
+            Warning: the `generate-lockfiles` goal does not yet work if you have local
+            requirements. Support is coming in a future Pants release.
```

Comment:
I removed the stuff about manual lock file generation because that will not support local file requirements either. 

I debated mentioning the workaround of hosting the files in a private index. Thoughts?